### PR TITLE
Cygwin path fix.

### PR DESCRIPTION
**Pull request by tsulej**
See error description [here](https://github.com/clojure-emacs/refactor-nrepl/issues/183).

Context: Emacs under Cygwin generates unix-like paths which are not accepted by nREPL. This error is visible when running `replace symbol`.
Cider provides two functions: `cider-to-nrepl-filename-function` and `cider-from-nrepl-filename-function` to make proper conversions. These functions make conversions only when you run Emacs under Cygwin.

Following fix wraps `(buffer-file-name)` when creating request.
To process middleware response I introduced `cljr--get-valid-filename` function which extracts `:file` from the hash and converts path to back to unix-like.

I tested only `replace symbol` case.

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [x] The new code is not generating byte compile warnings (run `cask exec emacs -batch -Q -L . -eval "(progn (setq byte-compile-error-on-warn t) (batch-byte-compile))" clj-refactor.el`)
- [x] All tests are passing (run `./run-tests.sh`)
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@
 - [#320](https://github.com/clojure-emacs/clj-refactor.el/issues/320) `*data-readers*` ignored when searching for macros.
 - [#339](https://github.com/clojure-emacs/clj-refactor.el/issues/339) Teach stop refer to understand multiline refer clauses.
 - [#341](https://github.com/clojure-emacs/clj-refactor.el/issues/341) Avoid creating circular dependencies with move form when source namespace refers to the target namespace with a require.
+- Cygwin path fix. Conversion between Unix-like path generated by Emacs and Windows-like path required by nREPL.
```
Comment (benedekfazekas) [inline CHANGELOG.md]: please reference the `refactor-nrepl` issue you created (see examples of this below in the changelog). also perhaps mention the fact that a cider solution is reused.

File: clj-refactor.el [hunk 2]
```diff
@@ -2401,8 +2401,12 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-promote-function
 root."
   (s-chop-prefix (cljr--project-dir) path))
 
+(defun cljr--get-valid-filename (hash)
+    "Get :file value from the hash table and convert path if necessary. Cygwin fix."
```
Comment (benedekfazekas) [inline clj-refactor.el]: I don't think "Cygwin fix". is needed here.

File: clj-refactor.el [hunk 3]
```diff
@@ -2350,7 +2350,7 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-promote-function
   (= (point) (cljr--point-after 'end-of-buffer)))
 
 (defun cljr--find-symbol-sync (symbol ns)
-  (let* ((filename (buffer-file-name))
+  (let* ((filename (funcall cider-to-nrepl-filename-function (buffer-file-name)))
```
Comment (expez) [inline clj-refactor.el]: You can delete all occurrences of `funcall`.  If you know the name of the function you can call it like this:

`(cider-to-nrepl-filename-function arg-goes-here)`

**Comment by expez** [comment]
Have you tested all this manually?  We don't have any real integration tests testing the entire stack, unfortunately.

File: clj-refactor.el (hunk 3 above)
Comment (tsulej) [inline clj-refactor.el]: Unfortunately not. These two functions are defined as `defvar` and calling directly causes the error.

File: CHANGELOG.md (hunk 1 above)
Comment (tsulej) [inline CHANGELOG.md]: Sure!

File: clj-refactor.el (hunk 2 above)
Comment (tsulej) [inline clj-refactor.el]: ok

**Comment by tsulej** [comment]
Yes, I've tested this manually. Only one case `replace symbol`. If you can point me to other functions which require nrepl interaction with filename inside I'll test them either.
Fortunately it should be safe fix, outside cygwin `cider-to-nrepl-filename-function` and `cider-from-nrepl-filename-function` is `identity`. Updating code after review.

**Comment by benedekfazekas** [comment]
thanks for the quick and nice fix @tsulej

**Pull request merged**